openai.el

```diff
@@ -57,11 +57,36 @@
   (when openai--show-log
     (apply 'message fmt args)))
 
+(defun openai-key--auth-source ()
+  "Retrieve the OpenAI API key from auth-source."
+  (let ((auth-info (auth-source-search :max 1
+                                       :host "api.openai.com"
+                                       :require '(:user :secret))))
+    (if auth-info
+          (funcall (plist-get (car auth-info) :secret))
+	(error "OpenAI API key not found in auth-source"))))
+
 ;;
 ;;; Request
 
-(defvar openai-key ""
-  "Generated API key.")
+
+(defcustom openai-key ""
```

Comment:
We have discussed not using `defcustom` in #5.

> `openai-key` should not be defcustom. Many users commit their custom file and they could accidentally commit their tokens.

I am not 100% sure, but it sounds reasonable to me. 🤔

Reply:
right, I hadn't seen that until now, I'm not sure how much it's possible to protect against people commiting their keys, but I agree with the point about an api package not having `defcustoms` - I'll revert this bit.